addons/mail/static/src/models/attachment/attachment.js

```diff
@@ -49,6 +49,20 @@ function factory(dependencies) {
             return data2;
         }
 
+        /**
+         * @override
+         */
+        static create(data) {
+            const isMulti = typeof data[Symbol.iterator] === 'function';
+            const dataList = isMulti ? data : [data];
+            for (const data of dataList) {
+                if (!data.id) {
+                    data.id = getAttachmentNextTemporaryId();
+                }
```

Comment:
Bad practice to mutate params. You do this so that local id matches id correctly for temporary fields, but ideally it should be done in a better way.

Reply:
Identifying fields should be fully provided at `create`/`insert`. The only change that would make sense here is to let `getAttachmentNextTemporaryId` be called from the outside, or manage this temporary notion differently.

Reply:
let's keep an exception for this one for the time being